web/app/src/views/Home.vue

```diff
@@ -503,21 +523,32 @@ const toggleGroupCollapse = (groupName) => {
 }
 
 const initializeCollapsedGroups = () => {
-  // Get saved uncollapsed groups from localStorage
   try {
     const saved = localStorage.getItem('gatus:uncollapsed-groups')
     if (saved) {
       uncollapsedGroups.value = new Set(JSON.parse(saved))
+      return
     }
-    // If no saved state, uncollapsedGroups stays empty (all collapsed by default)
   } catch (e) {
     console.warn('Failed to parse saved uncollapsed groups:', e)
     localStorage.removeItem('gatus:uncollapsed-groups')
-    // On error, uncollapsedGroups stays empty (all collapsed by default)
+  }
+
+  if (!collapseByDefault.value) {
+    const groups = Object.keys(combinedGroups.value || {})
+    uncollapsedGroups.value = new Set(groups) // expanded by default
+  } else {
+    uncollapsedGroups.value = new Set() // collapsed by default
   }
 }
 
+watch(() => combinedGroups.value, () => {
+  if (!localStorage.getItem('gatus:uncollapsed-groups')) {
+    initializeCollapsedGroups()
```

Comment:
Was this (`initializeCollapsedGroups`) not called before? Or am I missing something?
Looks like `initializeCollapsedGroups` was previously dead code